pydantic_settings/main.py

```diff
@@ -27,6 +28,11 @@ class SettingsConfigDict(ConfigDict, total=False):
     secrets_dir: str | Path | None
 
 
+# Extend config_keys by pydantic settings config keys to
+# support setting config through class kwargs.
+config_keys |= set(SettingsConfigDict.__annotations__.keys())
```

Comment:
This seems super magical... Can you add a reference to the pydantic code that we are using here?

Reply:
I improved the comment and explained it more by mentioning the class and method name that the magic happen there on Pydantic.

Please let me know by comment or suggestion if you think it is not clear yet.